mailing_list/tests.py

```diff
@@ -0,0 +1,242 @@
+        result = mailing_list.sync_listserv_membership()
+
+        mock_add_members.assert_called_with(mailing_list, kwargs['add_members'])
+        mock_delete_members.assert_called_with(mailing_list, kwargs['delete_members'])
+        self.assertEqual(result, kwargs['members_count'])
+
+    def test_sync_listserv_membership_add_members_to_listserv(self):
+        self._call_sync_listserv_membership(
+            listserv_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu'
+            },
+            enrolled_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'david_bonner@harvard.edu',
+                'sapna_mysore@harvard.edu'
+            },
+            add_members={'david_bonner@harvard.edu', 'sapna_mysore@harvard.edu'},
+            delete_members=set(),
+            members_count=4
+        )
+
+    def test_sync_listserv_membership_delete_members_from_listserv(self):
+        self._call_sync_listserv_membership(
+            listserv_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'david_bonner@harvard.edu',
+                'sapna_mysore@harvard.edu'
+            },
+            enrolled_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu'
+            },
+            add_members=set(),
+            delete_members={'david_bonner@harvard.edu', 'sapna_mysore@harvard.edu'},
+            members_count=2
+        )
+
+    def test_sync_listserv_membership_add_and_delete_members_from_listserv(self):
+        self._call_sync_listserv_membership(
+            listserv_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'david_bonner@harvard.edu'
+            },
+            enrolled_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'sapna_mysore@harvard.edu'
+            },
+            add_members={'sapna_mysore@harvard.edu'},
+            delete_members={'david_bonner@harvard.edu'},
+            members_count=3
+        )
+
+    @override_settings(IGNORE_WHITELIST=True)
+    def test_sync_listserv_membership_unsubscribed(self):
+        self._call_sync_listserv_membership(
+            listserv_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu'
+            },
+            enrolled_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'david_bonner@harvard.edu',
+                'douglas_hall@harvard.edu'
+            },
+            add_members={'david_bonner@harvard.edu'},
+            delete_members=set(),
+            members_count=3
+        )
+
+    def test_sync_listserv_membership_whitelist_not_set(self):
+        self._call_sync_listserv_membership(
```

Comment:
Maybe my explanation wasn't clear, but by populating a new lookup table, I was talking about having code look at the existing values in the database, use those values to populate a new table, then write the corresponding id back to the original table.  Nowhere in the migration (modulo enums) should there be the actual data that goes into the database.

Think of as something that should know how to transform data that exists in an install of the database (and have as few assumptions about that data as possible), not as a way to avoid writing a data loading script/process.